### PR TITLE
Gradle Groovy Application/Library Project Creation Wizards

### PR DESCRIPTION
**Pull request by lkishalmi**
This one depends on #7257 and has to be rebased on that once merged.

Though it is just a low hanging fruit to add Groovy project creation.

image

image

### CODE REVIEW

**Comment by lkishalmi** [comment]
Should the category be "Groovy with Gradle"?

**Comment by mbien** [comment]
> Should the category be "Groovy with Gradle"?

would make sense I think.

**Comment by lkishalmi** [comment]
Screenshot from 2024-04-14 13-54-47

**Review by mbien — APPROVED** [review]
cool! not tested but code+screenshot makes sense to me

I assume this will properly activate the groovy cluster, right? ;)

File: groovy/gradle.groovy/nbproject/project.properties [hunk 1]
```diff
@@ -14,6 +14,7 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-javac.source=1.8
+javac.source=17
+javac.target=17
```
Comment (mbien) [inline groovy/gradle.groovy/nbproject/project.properties:18]: if you refresh this PR, this has to become `javac.release=17`

File: groovy/gradle.groovy/manifest.mf [hunk 2]
```diff
@@ -1,6 +1,8 @@
 Manifest-Version: 1.0
 AutoUpdate-Show-In-Client: false
 OpenIDE-Module: org.netbeans.modules.gradle.groovy
+OpenIDE-Module-Layer: org/netbeans/modules/gradle/groovy/layer.xml
 OpenIDE-Module-Localizing-Bundle: org/netbeans/modules/gradle/groovy/Bundle.properties
+OpenIDE-Module-Java-Dependencies: Java > 17
```
Comment (mbien) [inline groovy/gradle.groovy/manifest.mf:6]: and this line should be removed

**Review by mbien — CHANGES_REQUESTED** [review]

File: groovy/gradle.groovy/manifest.mf (hunk 2 above)
Comment (lkishalmi) [reply groovy/gradle.groovy/manifest.mf:6]: Thanks!

**Comment by mbien** [comment]
seems to work. Although on first attempt it didn't seem to work. Since the wizard doesn't check if the project already exists, so I created a new groovy project but it opened the existing gradle project I had in /tmp and i was confused why it made a java project :)

**Review by mbien — APPROVED** [review]
sorry for the delay, looks good.

we have to add project name validation at some point though, since the wizard could overwrite someone's project like it happened to me during testing.

**Pull request merged**